extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -363,30 +366,41 @@ public void run()
           // The retrying business is because the KafkaConsumer throws OffsetOutOfRangeException if the seeked-to
           // offset is not present in the topic-partition. This can happen if we're asking a task to read from data
           // that has not been written yet (which is totally legitimate). So let's wait for it to show up.
-          final ConsumerRecords<byte[], byte[]> records = RetryUtils.retry(
-              new Callable<ConsumerRecords<byte[], byte[]>>()
-              {
-                @Override
-                public ConsumerRecords<byte[], byte[]> call() throws Exception
+          ConsumerRecords<byte[], byte[]> records = ConsumerRecords.empty();
+          try {
+            polling = true;
```

Comment:
I actually did an await on a condition first but was on the fence about which implementation was less intrusive or how much you wanted to keep the backoff algorithm in RetryUtils (which I don't think is that important for this case). If that seems cleaner to you I'm good with swapping the implementations.